tools/devops/automation/scripts/GitHub.psm1

```diff
@@ -268,6 +299,140 @@ class GitHubComments {
+        }
+
+        $query = @"
+query {
+    repository(owner:"$($this.Org)", name:"$($this.Repo)"){
+        pullRequest(number: $prID) {
+            comments(last: 100) {
+                edges {
+                    node {
+                        id
+                        isMinimized
+                        body
+                        author {
+                            login
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+"@
+        $payload = @{
+            query=$query
+        }
+        $body = ConvertTo-Json $payload
+        $response = Invoke-RestMethod -Uri $url -Headers $headers -Method "POST" -Body $body
+        # loop over the result and remove all the extra noise we are not interested in
+        $comments = [System.Collections.ArrayList]@()
+        foreach ($edge in $response.data.repository.pullRequest.comments.edges) {
+            $commentId = $edge.node.id
+            $isMinimized = $edge.node.isMinimized
+            $author = $edge.node.author.login
+            $body = $edge.node.body
+            $comments.Add([GitHubComment]::new($commentId, $author, $body, $isMinimized))
+        }
+        # at this point, we have the comments for the PR, but not the comments of the commits of the PR, yes, confusing. The github UI
+        # contains 2 sets of comments:
+        # 1. Comments on a PR (which is an issue really)
+        # 2. Comments on the commits that are part of the PR
+        #
+        # We are missing those in 2, we can get those with a second query
+
+        $query = @"
+query{
+    repository(owner:"$($this.Org)", name:"$($this.Repo)") {
+        pullRequest(number: $prID){
+            commits(last:100) {
+                edges {
+                    node {
+                        commit {
+                            ... on Commit {
+                                oid
+                                message
+                            } 
+                            comments (last:100) {
+                                edges {
+                                    node {
+                                        id
+                                        body
+                                        author {
+                                            login
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+"@
+        $payload = @{
+            query=$query
+        }
+        $body = ConvertTo-Json $payload
+        $response= Invoke-RestMethod -Uri $url -Headers $headers -Method "POST" -Body $body
+        foreach ($edge in $response.data.repository.pullRequest.commits.edges) {
+            # at this point a node is a commit, which has the following:
+            # commit
+            #   oid
```

Comment:
I see 'oid' here so maybe the other one is intentional

Reply:
It is, refers to object id